model/task/task.go

```diff
@@ -1745,6 +1740,108 @@ func (t *Task) Archive() error {
 	return nil
 }
 
+func ArchiveMany(tasks []Task) error {
+	if len(tasks) == 0 {
+		return nil
+	}
+	// add execution tasks of display tasks passed in, if they are not there
+	execTaskMap := map[string]*Task{}
+	for _, t := range tasks {
+		execTaskMap[t.Id] = &t
+	}
+	additionalTasks := []string{}
+	for _, t := range tasks {
+		for _, et := range t.ExecutionTasks {
+			if execTaskMap[et] == nil {
+				additionalTasks = append(additionalTasks, t.ExecutionTasks...)
+				continue
+			}
+		}
+	}
+	if len(additionalTasks) > 0 {
+		toAdd, err := Find(ByIds((additionalTasks)))
+		if err != nil {
+			return errors.Wrap(err, "unable to find execution tasks")
+		}
+		tasks = append(tasks, toAdd...)
+	}
+
+	archived := []interface{}{}
+	taskIds := []string{}
+	for _, t := range tasks {
+		archived = append(archived, *t.makeArchivedtask())
+		taskIds = append(taskIds, t.Id)
+	}
+
+	mongoClient := evergreen.GetEnvironment().Client()
+	ctx, cancel := evergreen.GetEnvironment().Context()
+	defer cancel()
+	session, err := mongoClient.StartSession()
+	if err != nil {
+		return errors.Wrap(err, "unable to start session")
+	}
+	defer session.EndSession(ctx)
+
+	txFunc := func(sessCtx mongo.SessionContext) (interface{}, error) {
+		oldTaskColl := evergreen.GetEnvironment().DB().Collection(OldCollection)
+		_, err = oldTaskColl.InsertMany(ctx, archived)
+		if err != nil {
+			return nil, err
+		}
+
+		taskColl := evergreen.GetEnvironment().DB().Collection(Collection)
+		_, err = taskColl.UpdateMany(ctx, bson.M{
+			IdKey: bson.M{
+				"$in": taskIds,
+			},
+		},
+			bson.M{
+				"$unset": bson.M{
+					AbortedKey:   "",
+					AbortInfoKey: "",
+				},
+			})
+		if err != nil {
+			return nil, err
+		}
+		_, err = taskColl.UpdateMany(ctx, bson.M{
+			IdKey: bson.M{
+				"$in": taskIds,
+			},
+			RestartsKey: bson.M{
```

Comment:
Why do we only increment restarts for tasks with restarts greater than 0? Don't all tasks start at 0?

Reply:
I preserved this because the singular archive function did it, but actually it's doing it wrong. I opened https://jira.mongodb.org/browse/EVG-13347 but would like to leave this as is for now